Various fixes / cleanup to SerialPort test suite

willdean opened the pull request:
This contains a number fairly minor changes, which are probably in three categories - cosmetic cleanup, fixes for problems introduced by me in the migration and fixes for problems in the original test suite which have only worked by luck in the past (changes like ReadLine.cs are in this last category)

If the project would prefer me to supply more, smaller, PRs I can do that - I am trying to strike some reasonable balance.

It's not worth calling-out comment typos on review at this stage, there are thousands.

JeremyKuhne commented on src/System.IO.Ports/tests/Legacy/SerialPort/DosDevices.cs:
```diff
@@ -161,5 +138,5 @@ private static char[] CallQueryDosDevice(string name, out int dataSize)
 
     [DllImport("Kernel32.dll", SetLastError = true, CharSet = CharSet.Auto)]
-    internal static extern int QueryDosDevice(string lpDeviceName, char[] lpTargetPath, int ucchMax);
+    static extern int QueryDosDevice(string lpDeviceName, char[] lpTargetPath, int ucchMax);
```
Should be explicit in all cases per our style guidelines

JeremyKuhne commented on src/System.IO.Ports/tests/Legacy/SerialPort/Event_Generic.cs:
```diff
@@ -152,7 +152,7 @@ public void EventHandler_ThrowsException()
     }
 
-    [ConditionalFact(nameof(HasNullModem))]
+    [ConditionalFact(nameof(HasNullModem), Skip="Always hangs")]
```
Should have an associated issue, we generally don't use Skip directly. The only exception would be a test that you're expected to manually run, which should have a detailed comment explaining that.

willdean commented:
@JeremyKuhne Thanks for the review - I've rebased the branch, addressed those two nits,  and force-pushed it - hopefully that hasn't sent everything spinning off into the weeds - I'll do a new PR if it has.

willdean commented:
Actually it appears I haven't rebased it at all, sigh.  Leave it with me for a minute

JeremyKuhne commented:
@dotnet-bot test this please (rerun to ensure stability)

JeremyKuhne commented:
@dotnet-bot test this please (rerun to ensure stability)

willdean commented:
@JeremyKuhne Is dotnet-bot a person or an automaton?  Is the protocol documented anywhere?

JeremyKuhne commented:
@willdean it is a bot, you can ask it to kick a specific run or all of them. Send it "help" to get the full list.

JeremyKuhne commented:
@willdean I fixed the conflict as it was just the comment spacing

willdean commented:
Thanks!

willdean commented:
@dotnet-bot test this please (rerun to ensure stability)

willdean commented:
@dotnet-bot help
